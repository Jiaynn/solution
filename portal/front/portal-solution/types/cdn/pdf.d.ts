declare let pdfExport: string

declare module '*.pdf' {
  export = pdfExport
}
