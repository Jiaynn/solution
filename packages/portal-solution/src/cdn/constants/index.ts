// 启用动态加速工单地址
export const enableDynamicURL = 'https://support.qiniu.com/tickets/new/form?category=%E5%85%B6%E4%BB%96%E7%B1%BB%E5%92%A8%E8%AF%A2&space=%E8%9E%8D%E5%90%88%20CDN'

// 证书配置提交工单 URL
export const certConfigHelpURL = 'https://support.qiniu.com/tickets/new/form?category=%E8%AF%81%E4%B9%A6%E9%97%AE%E9%A2%98&space=CDN'
