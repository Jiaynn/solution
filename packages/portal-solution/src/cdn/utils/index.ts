import { values } from 'lodash'
import { ReactNode } from 'react'
import { makeCancelled, UnknownException } from 'qn-fe-core/exception'

export function isPromise(obj: any): obj is Promise<any> {
  if (!obj) {
    return false
  }
  if (typeof obj === 'object' || typeof obj === 'function') {
    if (typeof obj.then === 'function') {
      return true
    }
  }
  return false
}

export interface IAntdStyleMethodOptions {
  title?: ReactNode
  content?: ReactNode
  onOk?: (...args: any[]) => void
  onCancel?: (...args: any[]) => void
}

export type AntdStyleMethod<T extends IAntdStyleMethodOptions> = (options: T) => void

export type PromiseStyleMethod<T, P> = (options: T) => Promise<P>

// 把 antd 中如 Modal.confirm 风格（onCancel, onOk）的接口转换成 promise 接口
export function antdToPromise<
  T extends IAntdStyleMethodOptions,
  M extends AntdStyleMethod<T>,
  P
>(method: M): PromiseStyleMethod<T, P> {
  return (options: T) => new Promise((resolve, reject) => {
    const onCancel = (...args: any[]) => {
      if (options.onCancel) {
        options.onCancel(...args)
      }
      reject(makeCancelled())
    }
    const onOk = (...args: any[]) => {
      if (options.onOk) {
        options.onOk(...args)
      }
      resolve(args[0])
    }
    method({
      ...(options as any),
      onCancel,
      onOk
    })
  })
}

function getLexicValue(char: string) {
  return char ? char.charCodeAt(0) : -1
}

export function lexicCompare(strA: string, strB: string) {
  for (let i = 0, len = Math.max(strA.length, strB.length); i < len; i++) {
    const vA = getLexicValue(strA.charAt(i))
    const vB = getLexicValue(strB.charAt(i))
    if (vA !== vB) {
      return vA - vB
    }
  }
  return 0
}

export function valuesOfEnum<T extends object>(target: T): Array<T[keyof T]> {
  const fakeKeys = values(target).filter(value => typeof value !== 'string').map(String)
  const realKeys = Object.keys(target).filter(key => fakeKeys.indexOf(key) === -1)

  return (realKeys as Array<keyof T>).map(key => target[key])
}

export function assertUnreachable(): never
export function assertUnreachable(v: never, message?: string): never
export function assertUnreachable(v?: never, message = "Didn't expect to get here") {
  throw new UnknownException(message + v)
}

/**
 * 过滤数组的 “空” 值，其中空值指的是：
 * - null
 * - undefined
 */
export function nonEmptyArray<T>(data: Array<T | undefined | null>): T[] {
  return data.filter((it): it is T => it != null)
}

export function booleanPredicate<T>(value: T): value is Exclude<T, false | null | undefined | '' | 0> {
  return Boolean(value)
}

export function assert(condition: any, msg?: string): asserts condition {
  if (!condition) {
    throw Error(msg)
  }
}
