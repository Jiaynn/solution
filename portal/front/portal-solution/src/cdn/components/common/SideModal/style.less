@import '~cdn/utils/style/legacy-variable.less';

.side-modal-wrapper {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;

  .side-modal-mask {
    position: absolute;
    width: 100%;
    height: 100%;
    background-color: hsla(0,0%,100%,.7);
  }

  .side-modal-container {
    position: absolute;
    top: 0;
    right: 0;
    height: 100%;
    display: flex;
    flex-direction: column;
    background-color: #fff;
    border-left: 1px solid @border-level-3-color;
    box-shadow: 0 10px 60px 0 rgba(0,0,0,.05);
  }

  @header-height: 56px;
  @header-font-size: @font-size-h6;
  @header-padding: (@header-height - @header-font-size) / 2;
  @horizontal-padding: 24px;
  @vertical-padding: @header-padding;

  .side-modal-header {
    flex: 0 0 auto;
    width: 100%;
    line-height: @header-height;
    padding: 0 @horizontal-padding;
    position: relative;
    font-size: @header-font-size;
    font-weight: @font-weight-title;
    border-bottom: 1px solid @border-level-2-color;

    .side-modal-close {
      display: block;
      width: @header-height;
      height: @header-height;
      position: absolute;
      top: 0;
      right: 0;

      text-align: center;
      cursor: pointer;
    }
  }
  .side-modal-body {
    flex: 1 1 auto;
    padding: @vertical-padding @horizontal-padding;
    overflow-y: auto;
  }
  .side-modal-footer {
    flex: 0 0 auto;
    padding: @vertical-padding @horizontal-padding;
    border-top: 1px solid @border-level-2-color;
    text-align: right;
  }
}
