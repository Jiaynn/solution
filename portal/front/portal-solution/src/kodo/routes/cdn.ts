/**
 * @file CDN route TODO: 站内外链与内链分开
 * @author yinxulai <me@yinxulai.com>
 */

// CDN 概览
export function getCDNOverviewPath() {
  return '/cdn/overview'
}

// CDN 业务的域名创建
export function getCDNCreateDomainPath(bucketName: string) {
  const query = bucketName ? `?bucket=${encodeURIComponent(bucketName)}&fixBucket` : ''
  return `/cdn/domain/create${query}`
}

// CDN 业务的域名列表
export function getCDNDomainListPath() {
  return '/cdn/domain'
}

export function getCDNDomainDetailPath(domain: string) {
  return `/cdn/domain/${domain}`
}

export function getCDNCreateBucketDomainPath(bucketName: string) {
  return `/cdn/domain/create?bucket=${bucketName}&fixBucket`
}

export function getCDNRefreshPath() {
  return '/cdn/refresh-prefetch'
}

export function getCDNRefreshLogsPath() {
  return '/cdn/refresh-prefetch?tab=showLogs'
}
