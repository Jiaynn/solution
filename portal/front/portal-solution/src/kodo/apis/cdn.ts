/**
 * @file config api
 * @description region 相关的接口
 * @author lizhifeng <lizhifeng@qiniu.com>
 */

import autobind from 'autobind-decorator'
import { injectable } from 'qn-fe-core/di'
import { ApiException } from 'qn-fe-core/client'

import { ProxyClientV2 } from 'portal-base/common/apis/proxy'

import { proxy } from 'kodo/constants/apis'

interface CdnBaseResponse {
  code: number
  error: string
  [k: string]: any
}

export interface RefreshCdnSurplus extends CdnBaseResponse {
  urlQuotaDay: number // url 刷新当日限额
  urlSurplusDay: number // url 刷新当日余量
  dirQuotaDay: number // 目录刷新当日限额
  dirSurplusDay: number // 目录刷新当日余量
  quotaDay: number // 预取当日限额
  surplusDay: number // 预取当日剩余量
}

interface RefreshCdnOptions {
  product: 'cdn'
  urls: string[]
}

export class CdnApiException extends ApiException {
  constructor(
    public code: number,
    public payload: CdnBaseResponse,
    message?: string,
    cause?: unknown
  ) {
    super(
      'RefreshPrefetchApiException',
      code,
      message,
      undefined,
      cause
    )
  }
}

const messageMap = new Map<number, string>([
  [400031, '无效的 URL'],
  [400034, '刷新域名个数超过限制'],
  [400037, 'URL 已在队列中，无需重复提交'],
  [400039, '请求次数过多']
])

function produceResult<T extends CdnBaseResponse>(payload: T) {
  if (payload && typeof payload === 'object' && 'code' in payload && payload.code !== 200) {
    const message = messageMap.get(payload.code) || payload.error
    throw new CdnApiException(payload.code, payload, message)
  }

  return payload
}

@autobind
@injectable()
export class CdnApis {
  constructor(
    private proxyClient: ProxyClientV2
  ) {}

  getRefreshCdnSurplus(): Promise<RefreshCdnSurplus> {
    return this.proxyClient.get<RefreshCdnSurplus>(proxy.refreshCdnSurplus, {})
      .then(produceResult)
  }

  async refreshCdn(params: RefreshCdnOptions): Promise<void> {
    await this.proxyClient.post<CdnBaseResponse>(proxy.refreshCdn, params)
      .then(produceResult)
  }
}
