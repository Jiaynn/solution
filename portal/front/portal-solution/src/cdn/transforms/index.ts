import { QueryParams } from 'qn-fe-core/utils'

export function atoi(numberText: string, valueForInvalidInput = 0): number {
  const parsed = parseInt(numberText, 10)
  return Number.isNaN(parsed) ? valueForInvalidInput : parsed
}

export function atof(numberText: string, valueForInvalidInput = 0): number {
  const parsed = parseFloat(numberText)
  return Number.isNaN(parsed) ? valueForInvalidInput : parsed
}

export function getIntValue(e: React.FormEvent<any>, valueForInvalidInput = 0) {
  return atoi((e.target as any).value, valueForInvalidInput)
}

export function getFloatValue(e: React.FormEvent<any>, valueForInvalidInput = 0) {
  return atof((e.target as any).value, valueForInvalidInput)
}

export function trimAndFilter(lines: string[]): string[] {
  return (lines || []).map(
    line => line.trim()
  ).filter(
    line => !!line
  )
}

export function splitLines(linesStr: string | null | undefined): string[] {
  if (!linesStr) {
    return []
  }
  return linesStr.split('\n')
}

export function filterQuery(query?: QueryParams) {
  if (!query) {
    return
  }
  return Object.keys(query).reduce(
    (filtered, key) => (
      (query || {})[key]
      ? { ...filtered, [key]: (query || {})[key] }
      : filtered
    ), {} as QueryParams
  )
}

export function humanizeOnoff(isOn: boolean) {
  return isOn ? '开启' : '关闭'
}
