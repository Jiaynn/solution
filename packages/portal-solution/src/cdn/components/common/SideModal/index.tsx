/*
 * @file 右侧滑出的浮层
 * @author nighca <nighca@live.cn>
 */

import React, { CSSProperties } from 'react'
import ReactDOM from 'react-dom'
import classNames from 'classnames'
import autobind from 'autobind-decorator'
import Transition from 'react-transition-group/Transition'
import Icon from 'react-icecream/lib/icon'
import Button from 'react-icecream/lib/button'
import PopupContainer from 'react-icecream/lib/popup-container'

import './style.less'

export interface ISideModalProps {
  visible: boolean
  title?: React.ReactNode
  width?: string
  zIndex?: number
  okBtnDisabled?: boolean
  parentElement?: HTMLElement
  okBtnText?: string
  cancelBtnText?: string
  onOk?: () => void
  onCancel?: () => void
  children?: any
}

// SideModal 插槽
export const sideModalRoot = 'side-modal-root'

function ModalHeader(props: {
  title: React.ReactNode
  onCancel?: () => void
}) {
  return (
    <div className="side-modal-header">
      {props.title}
      <span className="side-modal-close" onClick={props.onCancel}>
        <Icon type="close" />
      </span>
    </div>
  )
}

function ModalFooter(props: {
  okBtnDisabled?: boolean
  okBtnText?: string
  cancelBtnText?: string
  onOk?: () => void
  onCancel?: () => void
}) {
  return (
    <div className="side-modal-footer">
      <Button type="ghost" onClick={props.onCancel}>{props.cancelBtnText || '取消'}</Button>
      <Button type="primary" disabled={props.okBtnDisabled} onClick={props.onOk}>{props.okBtnText || '确认'}</Button>
    </div>
  )
}

const defaultContainerWidth = '50%'

const transitionDuration = 300

const wrapperDefaultStyle = {
  opacity: 0.4,
  transition: `opacity ${transitionDuration}ms ease-in-out`
}

const wrapperTransitionStyles = {
  entering: { opacity: 0.4 },
  entered: { opacity: 1 }
}

const containerDefaultStyle = {
  transform: 'translateX(100%)',
  transition: `transform ${transitionDuration}ms ease-in-out`
}

const containerTransitionStyles = {
  entering: { transform: 'translateX(100%)' },
  entered: { transform: 'translateX(0%)' }
}

export default class SideModal extends React.Component<ISideModalProps, { top: number }> {
  state = {
    top: 0
  }

  wrapperElement: HTMLElement | null = null

  @autobind updateWrapperElement(el: HTMLElement | null) {
    this.wrapperElement = el
  }

  previousParentStyle: CSSProperties | null = null

  getParentElement() {
    return (
      this.props.parentElement
      || (this.wrapperElement && this.wrapperElement.parentElement)
    )
  }

  handleShow() {
    const parentElement = this.getParentElement()
    if (!parentElement) {
      return
    }

    const parentStyle = parentElement.style
    this.previousParentStyle = {
      overflow: parentStyle.overflow,
      overflowY: parentStyle.overflowY as CSSProperties['overflowY']
    }
    parentStyle.overflow = 'hidden'
    parentStyle.overflowY = 'hidden'
    this.setState({
      top: parentElement.scrollTop
    })
  }

  handleHide() {
    const previousParentStyle = this.previousParentStyle
    const parentElement = this.getParentElement()
    if (!parentElement || !previousParentStyle) {
      return
    }

    const parentStyle = parentElement.style
    parentStyle.overflow = previousParentStyle.overflow!
    parentStyle.overflowY = previousParentStyle.overflowY!
    this.previousParentStyle = null
  }

  componentDidUpdate(prevProps: ISideModalProps) {
    if (!prevProps.visible && this.props.visible) {
      this.handleShow()
    }
    if (prevProps.visible && !this.props.visible) {
      this.handleHide()
    }
  }

  @autobind
  updateDimensions() {
    if (this.props.visible) {
      const parentElement = this.getParentElement()
      this.setState({
        top: parentElement!.scrollTop
      })
    }
  }

  componentDidMount() {
    if (this.props.visible) {
      this.handleShow()
    }
    window.addEventListener('resize', this.updateDimensions)
  }

  componentWillUnmount() {
    if (this.props.visible) {
      this.handleHide()
    }
    window.removeEventListener('resize', this.updateDimensions)
  }

  @autobind handleMaskClick() {
    if (this.props.onCancel) {
      this.props.onCancel()
    }
  }

  render() {
    const props = this.props
    const wrapperClassName = classNames({
      'side-modal-wrapper': true,
      'side-modal-wrapper-visible': props.visible
    })

    const zIndex = props.zIndex != null ? props.zIndex : 10
    const top = this.state.top + 'px'

    const sideModalContent = (
      <Transition in={props.visible} timeout={transitionDuration} mountOnEnter unmountOnExit>
        {state => (
          <div className={wrapperClassName}
            ref={this.updateWrapperElement}
            style={{
              zIndex,
              top,
              ...wrapperDefaultStyle,
              ...wrapperTransitionStyles[state as keyof typeof wrapperTransitionStyles]
            }}
          >
            <div className="side-modal-mask" onClick={this.handleMaskClick}></div>
            <div className="side-modal-container"
              style={{
                width: props.width || defaultContainerWidth,
                ...containerDefaultStyle,
                ...containerTransitionStyles[state as keyof typeof wrapperTransitionStyles]
              }}
            >
              <ModalHeader title={props.title || ''} onCancel={props.onCancel} />
              <div className="side-modal-body">
                <PopupContainer>{props.children}</PopupContainer>
              </div>
              <ModalFooter
                okBtnDisabled={props.okBtnDisabled}
                okBtnText={props.okBtnText}
                cancelBtnText={props.cancelBtnText}
                onOk={props.onOk}
                onCancel={props.onCancel}
              />
            </div>
          </div>
        )}
      </Transition>
    )

    const root = document.querySelector(`.${sideModalRoot}`)
    if (root) {
      return ReactDOM.createPortal(sideModalContent, root)
    }

    return sideModalContent
  }
}
