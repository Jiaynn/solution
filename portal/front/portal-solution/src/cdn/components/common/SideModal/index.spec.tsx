/*
 * @file cases for domain create component
 * @author nighca <nighca@live.cn>
 */

import React from "react";

import { RendererUtils as Renderer } from "cdn/test";

import SideModal from ".";

it("renders correctly", () => {
  const renderer = new Renderer();
  const tree = renderer
    .createWithAct(
      <SideModal visible>
        <div>foo</div>
      </SideModal>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly with visible: false", () => {
  const renderer = new Renderer();
  const tree = renderer
    .createWithAct(
      <SideModal visible={false}>
        <div>foo</div>
      </SideModal>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
