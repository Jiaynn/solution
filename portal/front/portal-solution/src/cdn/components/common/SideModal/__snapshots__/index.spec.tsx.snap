// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="side-modal-wrapper side-modal-wrapper-visible"
  style={
    Object {
      "opacity": 1,
      "top": "0px",
      "transition": "opacity 300ms ease-in-out",
      "zIndex": 10,
    }
  }
>
  <div
    className="side-modal-mask"
    onClick={[Function]}
  />
  <div
    className="side-modal-container"
    style={
      Object {
        "transform": "translateX(0%)",
        "transition": "transform 300ms ease-in-out",
        "width": "50%",
      }
    }
  >
    <div
      className="side-modal-header"
    >
      
      <span
        className="side-modal-close"
      >
        <i
          aria-label="icon: close"
          className="anticon anticon-close"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="close"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
            />
          </svg>
        </i>
      </span>
    </div>
    <div
      className="side-modal-body"
    >
      <div
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <div>
          foo
        </div>
      </div>
    </div>
    <div
      className="side-modal-footer"
    >
      <button
        className="ant-btn"
        onClick={[Function]}
        type="button"
      >
        <span>
          取消
        </span>
      </button>
      <button
        className="ant-btn ant-btn-primary"
        onClick={[Function]}
        type="button"
      >
        <span>
          确认
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`renders correctly with visible: false 1`] = `null`;
